### PR TITLE
SLING-9487 - adding logic & test to set default for Logfilepath in Ma…

### PR DESCRIPTION
**Pull request by DominikSuess**
…nagedServiceFactory

### CODE REVIEW

**Comment by bosschaert** [comment]
LGTM 👍

**Comment by kwin** [comment]
Switching to https://docs.osgi.org/specification/osgi.cmpn/7.0.0/service.component.html#service.component-component.property.types is more sustainable but is certainly more effort.

File: src/main/java/org/apache/sling/commons/log/logback/internal/config/LoggerManagedServiceFactory.java [hunk 1]
```diff
@@ -20,17 +20,23 @@
...
+    public static final String LOG_FILE_DEFAULT = "logs/error.log";
+
     public String getName() {
         return "Logger configurator";
     }
 
     public void updated(String pid, @SuppressWarnings("rawtypes") Dictionary configuration)
             throws org.osgi.service.cm.ConfigurationException {
         try {
+            if (configuration.get(LogConfigManager.LOG_FILE) == null) {
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/commons/log/logback/internal/config/LoggerManagedServiceFactory.java:37]: We should not assume that the configuration dictionary is mutable

Comment (DominikSuess) [reply src/main/java/org/apache/sling/commons/log/logback/internal/config/LoggerManagedServiceFactory.java:45]: @cziegeler  please recheck

**Comment by DominikSuess** [comment]
@bosschaert - currently checking the background of the failing assertion - the question is if our assumption on expected behavior is correct - so if not defining any appender is a scenario is a valid state or not.

**Comment by DominikSuess** [comment]
@bosschaert , @cziegeler - I was crosschecking the case with @chetanmeh  and the assertion is no longer accurate. It is expected that a logger being registered via an explicit configuration always has an appended attached while console appenders are registered through other mechanisms and handled explicitly, the assertion checks for a scenario that is no longer relevant and an artifact of the very initial implementation of this.

**Comment by bosschaert** [comment]
So does that mean that the failing test need to be updated?

**Comment by chetanmeh** [comment]
Yes the failing test assertion can be removed

**Comment by bosschaert** [comment]
The errors that show up in the build are:

```
Caused by: java.io.IOException: Error resolving artifact org.ops4j.pax.exam:pax-exam-inject:jar:3.5.0: Could not transfer artifact org.ops4j.pax.exam:pax-exam-inject:jar:3.5.0 from/to central (http://repo1.maven.org/maven2/): Error transferring file: Server returned HTTP response code: 501 for URL: http://repo1.maven.org/maven2/org/ops4j/pax/exam/pax-exam-inject/3.5.0/pax-exam-inject-3.5.0.jar from http://repo1.maven.org/maven2/org/ops4j/pax/exam/pax-exam-inject/3.5.0/pax-exam-inject-3.5.0.jar
```

If I go to http://repo1.maven.org/maven2/org/ops4j/pax/exam/pax-exam-inject/3.5.0/pax-exam-inject-3.5.0.jar I get:

```
501 HTTPS Required. 
Use https://repo1.maven.org/maven2/
More information at https://links.sonatype.com/central/501-https-required
```

**Comment by cziegeler** [comment]
That's an age old pax exam version - updating to a newer one should solve that problem

**Comment by bdelacretaz** [comment]
SLING-9085 has more info on the Maven/https issue, and in SLING-9111 it was fixed by upgrading to pax exam 4.13.2. Which might require a few changes to the tests which are currently using the much older V3.5.0.

**Comment by DominikSuess** [comment]
@cziegeler, @bdelacretaz  - bumped the pax exam version which caused a bit of trouble wrt tinybundles which doesn't work with the latest versions of everything but the given combination seems to be the most recent that works with the latest tinybundles w/o refactorings.

**Pull request merged**